packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -33,12 +44,26 @@ export abstract class ContainerImage {
    */
   public abstract readonly imageName: string;
 
+  /**
+   * Optional credentials for a private image registry
+   */
+  public abstract readonly credentials?: RepositoryCreds;
+
   /**
    * Called when the image is used by a ContainerDefinition
    */
   public abstract bind(containerDefinition: ContainerDefinition): void;
+
+  /**
+   * Render the Repository credentials to the CloudFormation object
+   */
+  public renderRepositoryCredentials(): CfnTaskDefinition.RepositoryCredentialsProperty {
+    return {
+        credentialsParameter: this.credentials ? this.credentials.secret.secretArn : ''
```

Comment:
To avoid breaking all existing integration tests, maybe return undefined for this whole object if we don't have credentials?

```ts
public renderRepositoryCredentials(): CfnTaskDefinition.RepositoryCredentialsProperty  | undefined {
  if (!this.credentials) { return undefined; }
  return {
    credentialsParameter: this.credentials.secret.secretArn
  } ;
}
```

Reply:
And in fact, this method could be `abstract` here and only be implemented in `InternetHostedImage`, since that's the only image type that actually has any credentials at all.

Reply:
Well the others would need to do a base implementation of `undefined` as well.

Reply:
> To avoid breaking all existing integration tests...

Ah, the conditional here can probably be removed b/c this func should only be called if `credentials` is populated. See [renderContainerDefinition()](https://github.com/allisaurus/aws-cdk/pull/1/files#diff-541fb20ec37d9e651dd38abc923ccb58R382). Currently, the existing integ tests all pass.

> ...this method could be `abstract` here and be implemented in `InternetHostedImage`...

I tried returning `undefined` from this method before (as you show) and got yelled at by the compiler b/c it wasn't a return value of type `RepositoryCredentialsProperty`. So since I have to return a `RepositoryCredentialsProperty` obj regardless, I'd prefer to consolidate it into one func on ContainerImage. Is there a way around having to return an `RepositoryCredentialsProperty` in some cases?

Reply:
> got yelled at by the compiler

It's hard to see the change in the code I pasted because it scrolls off the side of the comment box, but you should have written the return type as:

```ts
CfnTaskDefinition.RepositoryCredentialsProperty  | undefined
```

That would have allowed you to return `undefined`.

Reply:
Ah, I totally missed that, thanks for pointing it out! In that case I'll update it per the above.